orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -0,0 +1,517 @@
+        lossful_model.graph.input[0].name: _to_numpy(copy.deepcopy(x)),
+        lossful_model.graph.input[1].name: _to_numpy(
+            copy.deepcopy(target).type(torch.int32)
+        ),
+    }
+
+    def crossentropy_loss(prediction, target):
+        loss = torch.nn.CrossEntropyLoss()
+        return loss(prediction, target)
+
+    # When
+    with tempfile.TemporaryDirectory() as onnx_dir_name:
+        onnx_file_path = os.path.join(onnx_dir_name, model_name)
+        onnx.save(lossful_model, onnx_file_path)
+        ort_session = onnxruntime.InferenceSession(
+            onnx_file_path, providers=C.get_available_providers()
+        )
+
+        ort_outs = ort_session.run(ort_output_names, ort_inputs)
+        torch_outs = crossentropy_loss(model(copy.deepcopy(x)), copy.deepcopy(target))
+
+        # Then
+        assert np.allclose(ort_outs[0], _to_numpy(torch_outs))
+
+
+def test_mse_loss_training_graph_execution():
+    # Given
+    device = "cuda"
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = SimpleNet(D_in, H, D_out).to(device)
+    x = torch.randn(N, D_in, device=device)
+    target = torch.randn(N, D_out, device=device)
+    onnx_model = _get_onnx_model(model, (copy.deepcopy(x),))
+    simple_graph = SimpleTrainingGraphWithMSELoss(onnx_model)
+    lossful_model = simple_graph()
+    model_name = "model_with_mse_loss.onnx"
+    ort_output_names = [lossful_model.graph.output[0].name]
+    for name, _ in model.named_parameters():
+        ort_output_names.append(f"{name}_grad.accumulation.out")
+    ort_inputs = {
+        lossful_model.graph.input[0].name: _to_numpy(copy.deepcopy(x)),
+        lossful_model.graph.input[1].name: _to_numpy(copy.deepcopy(target)),
+    }
+    for name, param in model.named_parameters():
+        ort_inputs[name] = _to_numpy(copy.deepcopy(param))
+        ort_inputs[f"{name}_grad.accumulation.buffer"] = _to_numpy(
+            torch.zeros_like(param)
+        )
+    ort_inputs["lazy_reset_grad"] = np.full(1, True)
+
+    def mse_loss(prediction, target):
+        loss = torch.nn.MSELoss()
+        return loss(prediction, target)
+
+    # When
+    with tempfile.TemporaryDirectory() as onnx_dir_name:
+        onnx_file_path = os.path.join(onnx_dir_name, model_name)
+        onnx.save(lossful_model, onnx_file_path)
+        ort_session = onnxruntime.InferenceSession(
+            onnx_file_path, providers=C.get_available_providers()
+        )
+
+        ort_outs = ort_session.run(ort_output_names, ort_inputs)
+        torch_outs = mse_loss(model(copy.deepcopy(x)), copy.deepcopy(target))
+        torch_outs.backward()
+
+        # Then
+        # assert loss is close
+        assert np.allclose(ort_outs[0], _to_numpy(torch_outs))
+
+        # assert all the gradients are close
+        for ort_grad, pt_param in zip(ort_outs[1:], model.parameters()):
+            assert np.allclose(ort_grad, _to_numpy(pt_param.grad))
+
+
+def test_crossentropy_loss_training_graph_execution():
+    # Given
```

Comment:
since a bunch of these tests share the same code, can you simplify this with helper functions